### ISSUE TITLE
Integration into rust-cv

### ISSUE THREAD

**Issue by vadixidav** [indianajohn/akaze-rust#6]
Hey, I am interested in if you would support integrating this crate with rust-cv. That would involve using rust-cv types like [`ImageKeyPoint`](https://docs.rs/cv-core/0.7.1/cv_core/struct.ImageKeyPoint.html). I have started doing [a little image matching](https://github.com/rust-cv/vslam-sandbox/blob/0847af1360dbc0dd3a94d7d712fb7a8a01a57088/src/main.rs#L40) with `rust-cv` crates such as [`arrsac`](https://crates.io/crates/arrsac), [`eight-point`](https://crates.io/crates/eight-point), [`cv-core`](https://crates.io/crates/cv-core), and [`sample-consensus`](https://crates.io/crates/sample-consensus).

Much of the binaries and examples you have today can be effectively implemented now on top of the rust-cv ecosystem, but there is more work that needs to be done. Specifically, I would like to create types for the descriptors and integrate them with matching via the [`hnsw`](https://crates.io/crates/hnsw) crate. If the `akaze` crate produced these descriptors (with SIMD alignment) directly, it would make the amount of glue logic necessary to perform matching much smaller. I can perform the work necessary to undergo this process. This would involve adding the descriptor types, creating abstractions around feature extraction (similar to [`Feature2D`](https://docs.opencv.org/master/d0/d13/classcv_1_1Feature2D.html)), and using the `ImageKeyPoint` type. We can also refactor the binaries out into various exectuables that support choosing from multiple feature extraction techniques at runtime so that we can benchmark, compare, and use them all together.

Let me know if you think this is a good idea. If so, I will get to work on it right away. If not, I can make an adapter crate to integrate it, but I will probably need to file a follow-up issue for refactoring the crate to reduce allocations since right now it performs a great deal of allocations. I will probably also need to work on SIMD integration since SIMD typically goes faster when all your memory is properly aligned. It also should support `#![no_std]` for embedded use cases, as I am trying to do that for all rust-cv crates.

If you are interested in moving the crate to the [rust-cv](https://github.com/rust-cv/) organization, feel free to. I am inviting you to the org right now as an owner, so if you accept you should have full privileges. Feel free to join the discord as well. We have a bunch of CV people on there at this point, including the maintainer of OpenMVG.

**Comment by vadixidav** [comment]
Before I start working on this in full (I already started working a little bit in my fork), I think I want to try and create common types for descriptors and doing kNN/ANN searches before revisiting this. This crate can then bring that in and use those descriptor types. I will do some experimentation on AKAZE once that is complete.
